stripe/src/main/java/com/stripe/android/view/CardFormView.kt

```diff
@@ -0,0 +1,419 @@
+
+    private val cardContainer = viewBinding.cardMultilineWidgetContainer
+
+    @VisibleForTesting
+    internal val cardMultilineWidget = viewBinding.cardMultilineWidget
+
+    @VisibleForTesting
+    internal val countryPostalDivider = viewBinding.countryPostalDivider
+
+    private val postalCodeContainer = viewBinding.postalCodeContainer
+
+    @VisibleForTesting
+    internal val errors = viewBinding.errors
+
+    private val postalCodeValidator = PostalCodeValidator()
+
+    @VisibleForTesting
+    internal val postalCodeView = viewBinding.postalCode
+
+    @VisibleForTesting
+    internal val countryLayout = viewBinding.countryLayout
+
+    private var formBackgroundColorStateList: ColorStateList? = null
+
+    private var style: Style = Style.STANDARD
+
+    private val cardErrors = FieldErrors()
+
+    enum class Style {
+        STANDARD,
+        BORDERLESS
+    }
+
+    /**
+     * Retrieve a [CardParams] representing the card details if all these fields are valid:
+     * card number, expiration date, CVC, postal, country.
+     * Returns null otherwise and display corresponding errors.
+     */
+    val cardParams: CardParams?
+        get() {
+            // first validate card fields
+            if (!cardMultilineWidget.validateAllFields()) {
+                cardMultilineWidget.shouldShowErrorIcon = true
+                return null
+            }
+            cardMultilineWidget.shouldShowErrorIcon = false
+
+            // then validate postal and country
+            if (!isPostalValid()) {
+                showPostalError()
+                return null
+            }
+
+            // validation OK, clear any error view
+            setErrorMessageAndToggleVisibility(null)
+
+            val expirationDate =
+                requireNotNull(cardMultilineWidget.expiryDateEditText.validatedDate)
+            val postalCode = postalCodeView.text?.toString()
+
+            return CardParams(
+                brand = cardMultilineWidget.brand,
+                loggingTokens = setOf(CARD_FORM_VIEW),
+                number = cardMultilineWidget.validatedCardNumber?.value.orEmpty(),
+                expMonth = expirationDate.month,
+                expYear = expirationDate.year,
+                cvc = cardMultilineWidget.cvcEditText.text?.toString(),
+                address = Address.Builder()
+                    .setCountry(countryLayout.selectedCountry.toString())
+                    .setPostalCode(postalCode)
+                    .build()
+            )
+        }
+
+    init {
+        setupCountryAndPostal()
+        setupCardWidget()
+
+        context.withStyledAttributes(
+            attrs,
+            R.styleable.StripeCardFormView
+        ) {
+            formBackgroundColorStateList =
+                getColorStateList(R.styleable.StripeCardFormView_formBackgroundColorStateList)
+            style = Style.values()[getInt(R.styleable.StripeCardFormView_cardFormStyle, 0)]
```

Comment:
Slight FUD about the enum value order being significant. Maybe update Style to include a `val attrValue: Int` property so we don't have to worry about a future order change breaking the integration?

```
enum class Style(
  internal val attrValue: Int
) {
  Standard(0),
  Borderless(1)
}
```